### PR TITLE
AcctIdx: flush inserts to disk

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
during bg flush, write dirty items to disk.
There is a placeholder here for removing items from in-mem after writing to disk. It has complexity, so I broke it into a separate pr.
#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/in_mem_accounts_index.rs [hunk 1]
```diff
@@ -419,6 +419,11 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
         self.storage.wait_dirty_or_aged.notify_one();
     }
 
+    fn should_remove_from_mem(&self, current_age: Age, entry: &AccountMapEntry<T>) -> bool {
+        // this could be tunable dynamically based on memory pressure
+        current_age == entry.age()
+    }
+
     fn flush_internal(&self) {
         let was_dirty = self.bin_dirty.swap(false, Ordering::Acquire);
```
Comment (brooksprumo) [inline runtime/src/in_mem_accounts_index.rs:428]: Not related to this PR, but this swap can probably be Relaxed. No need to change it now, just wanted to put it out there for us to be aware of and think about.

File: runtime/src/in_mem_accounts_index.rs [hunk 2]
```diff
@@ -428,21 +433,52 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
...
+            flush_entries_updated_on_disk += 1;
+            disk.insert(&k, (&v.slot_list.read().unwrap(), v.ref_count()));
         }
+        Self::update_time_stat(&self.stats().flush_update_us, m);
+        Self::update_stat(
+            &self.stats().flush_entries_updated_on_disk,
+            flush_entries_updated_on_disk,
+        );
+
+        // loop that processes 'removes' will go here
```
Comment (brooksprumo) [inline runtime/src/in_mem_accounts_index.rs:480]: I assuming this will be in a different PR, yes?

**Review by brooksprumo — APPROVED** [review]
lgtm, assuming that "future code here" section will land in a subsequent PR.

File: runtime/src/in_mem_accounts_index.rs (hunk 2 above)
Comment (jeffwashington) [reply runtime/src/in_mem_accounts_index.rs:480]: #20076

File: runtime/src/in_mem_accounts_index.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/in_mem_accounts_index.rs:428]: can you tell me your thinking?

Comment (jeffwashington) [reply runtime/src/in_mem_accounts_index.rs:428]: also in description:
> There is a placeholder here for removing items from in-mem after writing to disk. It has complexity, so I broke it into a separate pr.

File: runtime/src/in_mem_accounts_index.rs (hunk 2 above)
Comment (jeffwashington) [reply runtime/src/in_mem_accounts_index.rs:480]: also in description:
> There is a placeholder here for removing items from in-mem after writing to disk. It has complexity, so I broke it into a separate pr.

File: runtime/src/in_mem_accounts_index.rs (hunk 1 above)
Comment (brooksprumo) [reply runtime/src/in_mem_accounts_index.rs:428]: Sorry, I think I'm wrong... I must've gotten mixed up with the `v.dirty()` and `v.set_dirty()` later down in the loop...
